File: app/src/main/java/nz/co/codebros/quakesnz/interactor/SelectFeatureInteractorImpl.java
package nz.co.codebros.quakesnz.interactor;

import io.reactivex.Observer;
import nz.co.codebros.quakesnz.model.Feature;

/**
 * Created by leandro on 19/06/17.
 */

public class SelectFeatureInteractorImpl implements SelectFeatureInteractor {

    private final Observer<Feature> featureObserver;

    public SelectFeatureInteractorImpl(Observer<Feature> featureObserver) {
        this.featureObserver = featureObserver;
    }

    @Override
    public void execute(Feature feature) {
        featureObserver.onNext(feature);
    }
}

File: app/src/main/java/nz/co/codebros/quakesnz/interactor/SelectFeatureInteractorImpl.kt
package nz.co.codebros.quakesnz.interactor

import io.reactivex.Observer
import nz.co.codebros.quakesnz.core.data.Feature

/**
 * Created by leandro on 19/06/17.
 */

class SelectFeatureInteractorImpl(
        private val featureObserver: Observer<Feature>
) : SelectFeatureInteractor {
    override fun execute(feature: Feature) {
        featureObserver.onNext(feature)
    }
}
